ui/panels-plugin/src/plugins/trace-table/trace-table-model.ts

```diff
@@ -11,16 +11,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { MouseEvent } from 'react';
+
 /**
  * The Options object type supported by the TraceTable panel plugin.
  */
 // Note: The interface attributes must match cue/schemas/panels/trace-table/trace-table.cue
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface TraceTableOptions {}
+export interface TraceTableOptions {
+  onTraceClick?: (e: MouseEvent, traceId: string) => void;
```

Comment:
Is this fine? I think in general the `spec` of a Panel should be JSON-serializable, but in this case I need functions for
* running `setQueries()` on the explore page
* creating a link to the trace detail page (useful when the panel gets embedded outside of perses)

Reply:
Maybe it's better to do it in `TraceTableProps` and keep `TraceTableOptions` interface to model stuff?
```ts
export interface TraceTableProps extends PanelProps<TraceTableOptions> {
  onTraceClick?: (e: MouseEvent, traceId: string) => void;
  traceLink?: (traceId: string) => string;
}
``` 

Reply:
I like this idea :+1:

If these properties should be set, this panel (React component) must be embedded from code anyway, and it's not possible to set this property from a JSON dashboard specification.